nomos-services/data-availability/indexer/src/lib.rs

```diff
@@ -0,0 +1,420 @@
+    >,
+    ClPool::Item: Clone + Eq + Hash + Debug + 'static,
+    ClPool::Key: Debug + 'static,
+    DaPool::Item: Clone + Eq + Hash + Debug + 'static,
+    DaPool::Key: Debug + 'static,
+    A::Backend: 'static,
+    TxS: TxSelect<Tx = ClPool::Item>,
+    BS: BlobCertificateSelect<Certificate = DaPool::Item>,
+    DaStorage: DaStorageAdapter,
+    ConsensusStorage: StorageBackend + Send + Sync + 'static,
+{
+    service_state: ServiceStateHandle<Self>,
+    indexer: Indexer,
+    storage_relay: Relay<StorageService<DaStorage::Backend>>,
+    consensus_relay: ConsensusRelay<
+        A,
+        ClPool,
+        ClPoolAdapter,
+        DaPool,
+        DaPoolAdapter,
+        DaVerificationProvider,
+        TxS,
+        BS,
+        ConsensusStorage,
+    >,
+}
+
+pub enum DaMsg<B, V> {
+    AddIndex {
+        vid: V,
+    },
+    GetRange {
+        ids: Box<dyn Iterator<Item = V> + Send>,
+        reply_channel: Sender<Vec<B>>,
+    },
+}
+
+impl<B: 'static, V: 'static> Debug for DaMsg<B, V> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            DaMsg::AddIndex { .. } => {
+                write!(f, "DaMsg::Index")
+            }
+            DaMsg::GetRange { .. } => {
+                write!(f, "DaMsg::Get")
+            }
+        }
+    }
+}
+
+impl<B: 'static, V: 'static> RelayMessage for DaMsg<B, V> {}
+
+impl<
+        Indexer,
+        DaStorage,
+        Consensus,
+        A,
+        ClPool,
+        ClPoolAdapter,
+        DaPool,
+        DaPoolAdapter,
+        DaVerificationProvider,
+        TxS,
+        BS,
+        ConsensusStorage,
+    > ServiceData
+    for DataIndexerService<
+        Indexer,
+        DaStorage,
+        Consensus,
+        A,
+        ClPool,
+        ClPoolAdapter,
+        DaPool,
+        DaPoolAdapter,
+        DaVerificationProvider,
+        TxS,
+        BS,
+        ConsensusStorage,
+    >
+where
+    Indexer: DaIndexer,
+    Indexer::Blob: 'static,
+    Indexer::VID: 'static,
+    A: NetworkAdapter,
+    ClPoolAdapter: MempoolAdapter<Payload = ClPool::Item, Key = ClPool::Key>,
+    ClPool: MemPool<BlockId = HeaderId>,
+    DaPool: MemPool<BlockId = HeaderId>,
+    DaPoolAdapter: MempoolAdapter<Key = DaPool::Key>,
+    DaPoolAdapter::Payload: Certificate + Into<DaPool::Item> + Debug,
+    DaVerificationProvider: MempoolVerificationProvider<
+        Payload = DaPoolAdapter::Payload,
+        Parameters = <DaPoolAdapter::Payload as Certificate>::VerificationParameters,
+    >,
+    ClPool::Item: Clone + Eq + Hash + Debug + 'static,
+    ClPool::Key: Debug + 'static,
+    DaPool::Item: Clone + Eq + Hash + Debug + 'static,
+    DaPool::Key: Debug + 'static,
+    A::Backend: 'static,
+    TxS: TxSelect<Tx = ClPool::Item>,
+    BS: BlobCertificateSelect<Certificate = DaPool::Item>,
+    DaStorage: DaStorageAdapter,
+    ConsensusStorage: StorageBackend + Send + Sync + 'static,
+{
```

Comment:
If we can relax the bounds in the struct,probably we can relax them here as well.